spec/mongoid/association/embedded/embeds_many/proxy_spec.rb

```diff
@@ -2148,6 +2148,44 @@ class TrackingIdValidationHistory
         end
       end
     end
+
+    context "with block" do
+      let!(:author) do
+        Person.create!(title: 'Person')
+      end
+
+      let!(:video_one) do
+        author.videos.create!(title: 'video one')
+      end
+
+      let!(:video_two) do
+        author.videos.create!(title: 'video two')
+      end
+
+      it "finds one" do
+        expect(
+          author.videos.find do |video|
+            video.title == 'video one'
+          end
+        ).to eq(video_one)
+      end
+
+      it "finds multiple" do
```

Comment:
This actually returns a single document, right? Is the test meant to exercise the sitution when multiple documents match the block condition?

Reply:
Oh, my bad. We should expect multiple documents to be found here. Will fix and check again.

Reply:
Corrected.